app/controllers/blog_posts_controller.rb

```diff
@@ -1,6 +1,7 @@
 class BlogPostsController < ApplicationController
   def show
     @blog_post = authorize BlogPost.find(params[:id])
+    @featured_blog_post = FeaturedBlogPost.find_or_initialize_by(blog_post: @blog_post)
```

Comment:
how come we are doing a `find_or_initialize_by`, shouldn't this exist already?

Reply:
So the post that's being shown may not be "featured", in which case a `FeaturedBlogPost` will be initialized here. If it is featured, then that will be looked up.

This basically sets up the data so that I can render buttons for admin users to "feature" or "remove feature" on the post. Pundit policies want to have an object, which is why it's nice to initialize one if the blog hasn't been featured yet. I can still say `if policy(@featured_blog_post).create?` and have things work nicely. Hopefully that makes sense.

Reply:
hmmm still seems kind of strange. Like can't we just force the policy check to use a specific policy (i.e: the featured one for that blog post). As for the check, can't we just do like an `includes(:featured_blog_post)` and then have a method like if `blog_post.featured?` or something?

Reply:
Hmm yeah that `featured?` method idea sounds better. I'll cook up a refactor real quick.

Reply:
Alright managed to find enough time tonight to get some changes in. I ended up still being able to call the policies normally anyway. The `#featured?` method is good though and in hindsight, probably something I should've considered 😵‍💫

Reply:
it's all good! We are all here to learn and glad I could teach you something new.